### PR TITLE
CMakeRefactor

### PR DESCRIPTION
**Pull request by szymon-filipiak**
So this PR is more of a code health improvement. Please let me know if you have any additional questions about code there. 
It probably contains an error that causes only the last line of the input file to be read. Couldn't find the root cause of it.

### CODE REVIEW

File: header/Text/TextReaderFactory.h [hunk 1]
```diff
@@ -5,9 +5,6 @@
 #ifndef GPU_KARP_RABIN_TEXTREADERFACTORY_H
 #define GPU_KARP_RABIN_TEXTREADERFACTORY_H
 
-#include <fstream>
```
Comment (WiktorPieklik) [inline header/Text/TextReaderFactory.h:8]: Maybe lack of this particular library is causing strange problem with reading only the last line of the input file.

**Review by WiktorPieklik — APPROVED** [review]
I have already tested this branch on my computer and everything works as expected without having this strange problem with reading only last line of input file.
Check my comment, maybe it's causing strange behavior on your computer.
Besides that, great job!

File: header/Text/TextReaderFactory.h (hunk 1 above)
Comment (szymon-filipiak) [reply header/Text/TextReaderFactory.h:8]: This header is included in TextFileReader.h file. It should be sufficient as TextFileReader is the only class that realy uses the file streams.

Comment (WiktorPieklik) [reply header/Text/TextReaderFactory.h:8]: I missed that, sorry. Ignore this comment.

**Review by zawadzki-dawid — APPROVED** [review]

**Pull request merged**